app/Http/Controllers/BeatmapsController.php

```diff
@@ -52,29 +53,55 @@ public function scores($id)
 
         $beatmap = Beatmap::findOrFail($id);
         $mode = Request::input('mode', Beatmap::modeStr($beatmap->playmode));
+        $enabled_mods = Request::input('enabledMods');
 
         try {
-            $scores = $beatmap
+            $query = $beatmap
                 ->scoresBest($mode)
                 ->defaultListing()
                 ->with('user');
         } catch (\InvalidArgumentException $ex) {
             return error_popup($ex->getMessage());
         }
 
+        if ($enabled_mods) {
+            if (!$user->isSupporter()) {
+                return error_popup(trans('errors.supporter_only'));
+            }
+
+            $query->whereIn('enabled_mods', [$enabled_mods, 0]);
```

Comment:
First of all, [this](https://puu.sh/qFzgY.png).

Also, I'm not sure if we can show scores with a *particular* mod enabled in any other way than iterating over the scores and checking the bitset on each one.